include/MiniFB_ex.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "MiniFB_ex_enums.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+    int mfb_open_ex(const char* name, int width, int height, int flags);
+    eBool mfb_set_viewport(unsigned offset_x, unsigned offset_y, unsigned width, unsigned height);
+
+    typedef void(*mfb_active_func)(eBool isActive);
+    typedef void(*mfb_resize_func)(int width, int height);
+    typedef void(*mfb_keyboard_func)(eKey key, eKeyMod mod, eBool isPressed);
+    typedef void(*mfb_char_input_func)(unsigned int code);
+    typedef void(*mfb_mouse_btn_func)(eMouseButton button, eKeyMod mod, eBool isPressed);
+    typedef void(*mfb_mouse_move_func)(int x, int y);
+    typedef void(*mfb_mouse_scroll_func)(eKeyMod mod, float deltaX, float deltaY);

```

Comment:
I think it would be nice to allow the user to just do (if `open_ex` is changed) to do `get_mouse_pos(window)` to not require callbacks, callbacks is a bit of a double edge sword when it comes to control flow of a program.

Reply:
Also if we really want callbacks all of these functions needs to have a `void* user_data` to allow the user to send in some data that can be used in the callback code. Otherwise it forces the user to have "state" globally which is bad for number of reasons.

Reply:
We could add "direct / poll" access and have both. So the users could choose the best option for them. Personally I prefer callbacks instead of polling.

Do you mean, adding a set_user_data function, and then pass this user_data to every callback? 
Yes, it would be nice.